ipaclient/install/client.py

```diff
@@ -3902,7 +3943,47 @@ def __init__(self, **kwargs):
                     "--all-ip-addresses")
 
 
+@group
+class PKINITInstallInterface(service.ServiceInstallInterface):
+    description = "PKINIT"
+
+    pkinit_identity = knob(
+        type=str,
+        default=None,
+        description=(
+            "PKINIT identity information (for example "
+            "FILE:/path/to/cert.pem,/path/to/key.pem)"
+        ),
+        cli_metavar="IDENTITY",
+    )
+
+    @pkinit_identity.validator
+    def pkinit_identity(self, value):
+        if not value.startswith(("FILE:", "PKCS11:", "PKCS12:")):
```

Comment:
One additional comment. I checked MIT Kerberos source code and the following prefixes are supported: "FILE:", "PKCS11:", "PKCS12:", "DIR:", and "ENV:".  They all documented in `krb5.conf(5)`, section "Specifying PKINIT identity information": https://github.com/krb5/krb5/blob/master/doc/admin/conf_files/krb5_conf.rst#specifying-pkinit-identity-information

Reply:
I have added `DIR:` support. `ENV:` is kinda pointless for our use case.

`pkinit_anchor` now supports `DIR:` and requires either `DIR:` or `FILE:`. Other prefixes are not supported by KRB5.

Reply:
Certainly, there could be configurations where people pass the reference to the bundle through the environment variable before calling out to ipa-client-install. Automation is fun. ;)

Reply:
If you insist on ENV ... here you are.